src/parquet/properties.h

```diff
@@ -38,6 +38,121 @@ struct ParquetVersion {
 static int64_t DEFAULT_BUFFER_SIZE = 0;
 static bool DEFAULT_USE_BUFFERED_STREAM = false;
 
+// should find a better name???
+class PARQUET_EXPORT EncryptionProperties {
+ private:
+  static inline uint8_t* str2bytes(std::string str) 
+  {
+    if (str.empty()) return nullptr;
+
+    char* cbytes = const_cast<char*>(str.c_str());
+    return reinterpret_cast<uint8_t*>(cbytes);
+  }
+
+ public:
+  EncryptionProperties() = default;
+  EncryptionProperties(Encryption::type algorithm, std::string key,
+                      std::string key_metadata, std::string aad) 
+  : algorithm_(algorithm), key_(key), key_metadata_(key_metadata), aad_(aad) {}
+
+  int key_length() { return static_cast<int>(key_.length()); }
+  uint8_t* key_bytes() { return str2bytes(key_); }
+
+  int aad_length() { return static_cast<int>(aad_.length()); }
+  uint8_t* aad_bytes() { return str2bytes(aad_); }
+
+  Encryption::type algorithm() { return algorithm_; }
+
+  std::string key_metadata() { return key_metadata_; }
+
+  std::string key() { return key_; }
+
+ private:
+  std::string key_;
```

Comment:
It's possible for another part of the application (e.g. user plugins running in the process) to gain access to keys by allocating a chunks of memory after a class like this as been destroyed.  Consider explicitly zeroing out strings containing secrets in the destructor of the containing object.